linera-execution/src/system.rs

```diff
@@ -799,7 +799,9 @@ where
         self.check_required_applications(&application_description, Some(&mut txn_tracker))
             .await?;
 
-        txn_tracker.add_created_blob(Blob::new_application_description(&application_description));
+        let blob = Blob::new_application_description(&application_description);
+        self.used_blobs.insert(&blob.id())?;
+        txn_tracker.add_created_blob(blob);
```

Comment:
Alternatively we could add all blobs at the end of the transaction, from the transactiont tracker. Rather than having this logic in multiple places.